task_manager/task_manager/tasks/mission.py

```diff
@@ -81,15 +81,13 @@ def execute_cb(self, goal_handle: ServerGoalHandle) -> MissionAction.Result:
             mission_result.skipped = False
 
             if subtask_result.task_status != TaskStatus.DONE:
-                if subtask.allow_skipping:
+                if subtask.allow_skipping and not goal_handle.is_cancel_requested:
                     mission_result.skipped = True
                     continue
 
-                if subtask_result.task_status == TaskStatus.CANCELED and goal_handle.is_cancel_requested:
-                    # Need to also check if the cancel was requested. If the goal was cancelled
-                    # through a system task, we cannot set the status to be cancelled and must abort instead.
```

Comment:
Previously the purpose of this comment was to point out a shortcoming, that if a subtask was cancelled, we cannot set the Mission end status to CANCELLED, unless the Mission was also cancelled.

I think it would be a good to have a mention about this somewhere, in a more understandable format. Especially now in this case that we are cancelling both a Mission and a subtask when the STOP task is called, and if the subtask is the one that is being cancelled first, we will end up having Mission with the ERROR status.

Reply:
Tried to add new comments explaining the behavior on this if else block :+1: